Common/src/main/java/at/petrak/hexcasting/api/casting/mishaps/MishapBadOffhandItem.kt

```diff
@@ -9,23 +9,23 @@ import net.minecraft.world.InteractionHand
 import net.minecraft.world.item.DyeColor
 import net.minecraft.world.item.ItemStack
 
-class MishapBadOffhandItem(val item: ItemStack, val hand: InteractionHand?, val wanted: Component) : Mishap() {
+class MishapBadOffhandItem(val item: ItemStack?, val wanted: Component) : Mishap() {
     override fun accentColor(ctx: CastingEnvironment, errorCtx: Context): FrozenPigment =
         dyeColor(DyeColor.BROWN)
 
     override fun execute(env: CastingEnvironment, errorCtx: Context, stack: MutableList<Iota>) {
         env.mishapEnvironment.dropHeldItems()
     }
 
-    override fun errorMessage(ctx: CastingEnvironment, errorCtx: Context) = if (item.isEmpty)
+    override fun errorMessage(ctx: CastingEnvironment, errorCtx: Context) = if (item?.isEmpty != false)
         error("no_item.offhand", wanted)
     else
         error("bad_item.offhand", wanted, item.count, item.displayName)
```

Comment:
I think rearranging it like this would make the intent a lot clearer:
```suggestion
    override fun errorMessage(ctx: CastingEnvironment, errorCtx: Context) = if (item?.isEmpty == false)
        error("bad_item.offhand", wanted, item.count, item.displayName)
    else
        error("no_item.offhand", wanted)
```